src/ArgParse/Basic.purs

```diff
@@ -426,7 +426,7 @@ choose name parsers =
           if next.saturated then
             ArgMatch
               (failDup help DuplicateArg $ ArgFold
-                { step: go1 (foldr (:) (ArgFold next : ams) acc)
+                { step: go1 (acc <> (ArgFold next : ams))
```

Comment:
Oh! Actually I think this is a bug. It's supposed to be `foldl`. The `acc` is a reversed list, and it needs to be reversed back to get the original order.

Reply:
Welp! Closing this PR then. I might open an issue later.

P.S I really think this project needs some tests.

Reply:
Indeed.